src/solvers/refinement/string_refinement.cpp

```diff
@@ -288,6 +289,22 @@ bool string_refinementt::add_axioms_for_string_assigns(
   return true;
 }
 
+template<typename T>
+T expr_cast(const exprt&);
+
+template<>
+std::size_t expr_cast<std::size_t>(const exprt& val_expr)
```

Comment:
Already documented the template itself - `T expr_cast` template (just above that snippet).